contracts/interchain-token-service/src/state.rs

```diff
@@ -170,6 +170,27 @@ pub fn load_chain_config(
         .ok_or_else(|| report!(Error::ChainNotFound(chain.to_owned())))
 }
 
+pub fn load_chain_configs(storage: &dyn Storage) -> Result<Vec<msg::ChainConfigResponse>, Error> {
+    let configs = CHAIN_CONFIGS
+        .range(storage, None, None, Order::Ascending)
+        .map(|res| {
+            res.change_context(Error::Storage)
+                .map(|(chain, config)| msg::ChainConfigResponse {
+                    chain,
+                    its_edge_contract: config.its_address,
+                    truncation: msg::TruncationConfig {
+                        max_uint: config.truncation.max_uint,
+                        max_decimals_when_truncating: config
+                            .truncation
+                            .max_decimals_when_truncating,
+                    },
+                    frozen: config.frozen,
```

Comment:
I would suggest to extract the mapping into it's own function, it's a separate concern to convert it into a msg type from loading it from storage. In fact, I think it's cleaner to return the actual chain configs here, and convert it in a separate step, at the call site